src/main/java/core/basesyntax/FileWork.java

```diff
@@ -8,7 +15,28 @@
  * Результат: web wide width world</p>
  */
 public class FileWork {
-    public String[] readFromFile(String fileName) {
-        return null;
+    private static final String LETTER = "w";
+
+    public String[] readFromFile(String fileName) throws RuntimeException {
+
+        try {
+            List<String> listOfLines = Files.readAllLines(Paths.get(fileName));
+            List<String> resultList = new ArrayList<>();
+
+            for (String line : listOfLines) {
+                String[] arrayOfWords = line.split(" ");
+                for (String word : arrayOfWords) {
+                    if ((word.substring(0, 1)).equalsIgnoreCase(LETTER)) {
+                        resultList.add(word.replaceAll("[^A-Za-zА-Яа-я0-9]", "").toLowerCase());
+                    }
+                }
+            }
+            Collections.sort(resultList);
+            return resultList.toArray(new String[resultList.size()]);
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
let's rethrow this as a RuntimeException

Reply:
done!